kj-rs/own.rs

```diff
@@ -0,0 +1,183 @@
+//! The `workerd-cxx` module containing the [`KjOwn<T>`] type, which is bindings to the `kj::Own<T>` C++ type
+use crate::fmt::display;
+use std::ffi::c_void;
+use std::fmt::{self, Debug, Display};
+use std::marker::PhantomData;
+use std::mem::MaybeUninit;
+use std::ops::Deref;
+use std::ops::DerefMut;
+use std::pin::Pin;
+
+/// A [`KjOwn<T>`] represents the `kj::Own<T>`. It is a smart pointer to an opaque C++ type.
+#[repr(C)]
+pub struct KjOwn<T>
```

Comment:
I like the pattern of putting everything that is shared with c++ in layout into `mod repr { ... }` to make it explicit in the name.